staging/src/k8s.io/apiserver/pkg/endpoints/handlers/rest_test.go

```diff
@@ -604,6 +618,52 @@ func TestPatchResourceWithRacingVersionConflict(t *testing.T) {
 	tc.Run(t)
 }
 
+func TestPatchResourceWithInvalidJSONPatch(t *testing.T) {
```

Comment:
It seems you need to change the `patchTestCase` framework a bit to make it work for your two test cases.

Instead of making that framework more complicated, how about making a focused test for `applyJSPatch`? You don't need to fake too much to make it work.

And you can move the tests to right below [TestPatchInvalid](https://github.com/kubernetes/kubernetes/blob/d472a5477705317706b09afc9589bb75496aeb02/staging/src/k8s.io/apiserver/pkg/endpoints/handlers/rest_test.go#L104). (And you can rename `TestPatchInvalid` to `TestStrategicMergePatchInvalid` to make is clear).

What do you think?

Reply:
Done.